src/client/client.cc

```diff
@@ -709,37 +872,64 @@ namespace Pistache::Http::Experimental
             if (totalBytes >= max_buffer)
             {
                 auto new_max_buffer = (max_buffer * 2);
-                char * new_buffer = 0;
-                if ((new_max_buffer > (8*1024*1024)) ||//new_max_buffer > 8MB ?
-                    (0 == (new_buffer = new char[new_max_buffer+16])))
+                char* new_buffer    = 0;
+                if ((new_max_buffer > max_max_buffer) || (nullptr == (new_buffer = new char[new_max_buffer + 16])))
                 {
-                    if (new_max_buffer > 268435456)
+                    if (new_max_buffer > max_max_buffer)
                         PS_LOG_WARNING("Receive buffer would be too big");
                     else
                         PS_LOG_WARNING_ARGS("Failed to alloc %d bytes memory",
-                                            new_max_buffer+16);
+                                            new_max_buffer + 16);
 
                     connection->handleResponsePacket(buffer, totalBytes);
                     break;
                 }
                 std::memcpy(new_buffer, buffer, max_buffer);
                 buffer_uptr = std::unique_ptr<char[]>(new_buffer);
-                buffer = new_buffer;
-                max_buffer = new_max_buffer;
+                buffer      = new_buffer;
+                max_buffer  = new_max_buffer;
             }
         }
     }
 
     Connection::Connection(size_t maxResponseSize)
-        : fd_(PS_FD_EMPTY)
-        , requestEntry(nullptr)
+        : requestEntry(nullptr)
         , parser(maxResponseSize)
     {
         state_.store(static_cast<uint32_t>(State::Idle));
         connectionState_.store(NotConnected);
     }
 
-    void Connection::connect(const Address& addr)
+    void Connection::connect(Address::Scheme scheme,
+#ifdef PISTACHE_USE_SSL
+                             SslVerification sslVerification,
+#endif // PISTACHE_USE_SSL
+                             const std::string& domain,
+                             const std::string* page)
+    {
+        const Address addr(helpers::httpAddr(
+            domain,
+            (Address::Scheme::Https == scheme) ? 443 : 0, // default port
```

Comment:
nit

```suggestion
            (scheme == Address::Scheme::Https) ? 443 : 0, // default port
```

Reply:
Oh, why do prefer that way round @Tachi107? Since scheme is a non-const variable (parameter actually), and Address::Scheme::Https is a constant, I would think it best to do `(Address::Scheme::Https == scheme)` to avoid any risk of using a single `=` and so doing an assignment. Though I admit I'm not super consistent about that.

What's the argument for the other way around?